### ISSUE TITLE
vue language tools not working / bad performance

### ISSUE THREAD

**Issue by boernard** [vuejs/language-tools#3801]
## Summary
The volar extension is not usable because code hints don't load.

I have a small nuxt project (4 components, 2 pages) with the following dev dependencies in package.json

```
  "devDependencies": {
    "@nuxt/devtools": "latest",
    "@nuxtjs/tailwindcss": "^6.8.0",
    "@types/node": "^18.17.3",
    "autoprefixer": "^10.4.15",
    "nuxt": "^3.6.5",
    "postcss": "^8.4.28",
    "tailwindcss": "^3.3.3",
    "typescript": "^5.2.2",
    "volar": "^0.0.2",
    "volar-service-vetur": "latest"
  }
```

Whenever I hover over a component, the code hint stays in the state `Loading...` and never actually finishes loading. The CPU process for the volar extension goes up to >85%. Same result when I start writing a component and press `CMD + space`, it stays in the Loading state.

<img width="958" alt="grafik" src="https://github.com/vuejs/language-tools/assets/7024749/a4173999-62c4-4fed-b127-ee2f1a1c6bcd">

Also VSCode's status bar keeps indicating `Initializing JS/TS language features`

## System setup
Apple M2
Visual Studio Code 1.85
TypeScript Vue Plugin (Volar) v1.8.25
Vue Language Features (Volar) v1.8.25

for profiling I pulled the current master of https://github.com/vuejs/language-tools at [b8627e7](https://github.com/vuejs/language-tools/commit/b8627e701cc9ef056d9ecaa66f56cb83dfca4d20)

[CPU-20231213T170606.cpuprofile.zip](https://github.com/vuejs/language-tools/files/13663075/CPU-20231213T170606.cpuprofile.zip)

when I open `output` in the vscode window and go to `vue semantic server` I get the following output
[vueSemanticServer.txt](https://github.com/vuejs/language-tools/files/13663299/vueSemanticServer.txt)

## Additional info
When I installed the volar language extension in Sublime, also there it doesn't work. Node process goes up in CPU usage and then continously crashes.

**Comment by funny-cat-happy** [comment]
I have the same problem as you. And my intellisense have slowed down.

**Comment by huzaifahj** [comment]
Identical setup as OP. Intellisense is very slow and code actions have to be disabled to get a sane file saving time.

**Comment by machao7** [comment]
any feedback?

**Comment by DanielViberg** [comment]
Experiencing the same in larger components.

**Comment by s3xysteak** [comment]
The same issue too. I am using Vue.volar & Vue.vscode-typescript-vue-plugin & takeover mode, It is so slow in both `.ts .vue`, which make me feel so sad. But it works greet in `.js` with only Vue.volar and no takeover mode.

**Comment by douglasg14b** [comment]
Intelisense is a multi-second operation now, it's cripplingly bad.

**Comment by huzaifahj** [comment]
Is anyone looking into this? Vue could have the best feature set and this would still be a deal-breaker for any serious project in 2024.

**Comment by johnsoncodehk** [comment]
Thanks for your report, the performance issues caused by `vue semantic server` have been resolved in v2 via #3788.

**Comment by boernard** [comment]
For me it didn't get resolved... I quickly uninstalled volar, reinstalled the vue - official extensions, restarted and tried a couple of other things but still, the behaviour is the same.
I am on vscode 1.91.1
